wolox-books/src/app/register/register.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  registerForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.registerForm = this.formBuilder.group({
+      name: [''],
+      lastName: [''],
+      email: ['', Validators.compose([Validators.required, Validators.email])],
+      password: ['', Validators.compose([Validators.required, Validators.pattern('^(?=.*[A-Z])(?=.*[0-9]).*$')])],
+      confirmPassword: ['', Validators.required]
+    },
+    {
+      validator: this.passwordConfirmation('password', 'confirmPassword')
+    });
+  }
+
+  passwordConfirmation = (password, confirmPassword) => {
+    return formGroup => {
+      const userPassword = formGroup.get(password)
+      const confirmedPassword = formGroup.get(confirmPassword);
+
+      return userPassword.value !== confirmedPassword.value && confirmedPassword.setErrors({ passwordConfirmation: false })
+    }
+  }
+
+  errorMessage = controlName => {
+    if (this.registerForm.get(controlName).errors) {
+      if (this.registerForm.get(controlName).errors.required) {
+        return 'This field is required';
+      } else if (this.registerForm.get(controlName).errors.email) {
+        return 'Invalid email format';
+      } else if (this.registerForm.get(controlName).errors.pattern) {
+        return 'Must have at least one number and one capital letter';
+      } else if (!this.registerForm.get(controlName).errors.passwordConfirmation) {
+        return 'The passwords must match';
+      }
```

Comment:
Actually I rather omit switches and many if/else statements concatenated, sometime I wanted to transform this into a for-in/foreach like sentence, iterating over the control object and its keys (errors, error.email... and so on) but I had the sensation that it would get more extended in lines of code, but right now I'm thinking again would be better like that, I'll change it to iterate over the object with the condition inside every step 👍